packages/core/lib/events/DomainEventEmitter.ts

```diff
@@ -16,36 +23,52 @@ import type {
   SingleEventHandler,
 } from './eventTypes'
 
+export type DomainEventEmitterDependencies<SupportedEvents extends CommonEventDefinition[]> = {
+  eventRegistry: EventRegistry<SupportedEvents>
+  metadataFiller: MetadataFiller
+  logger: Logger
+  errorReporter: ErrorReporter
```

Comment:
we should provide some dummy reporter implementation that just logs errors in case we want to make this mandatory

Reply:
Mmm, we already have the logger there as mandatory and is doing exactly that, so I would say we can make it option

Reply:
Change applied, thank youuu 🙏